### PR TITLE
Adsk contrib/reorganize materialx render

### PR DESCRIPTION
**Pull request by feldstj**

### CODE REVIEW

File: source/MaterialXRenderHw/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+add_library(MaterialXRenderHw STATIC 
+    ${materialx_source} 
+    ${materialx_header}
+)
+
+if(MSVC)
+    target_link_libraries(
+	MaterialXRenderHw
+	MaterialXRender
+        Opengl32
```
Comment (bernardkwok) [inline source/MaterialXRenderHw/CMakeLists.txt]: GL is not required here.

File: source/MaterialXRenderGlsl/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+)
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/MaterialXRenderGlsl/
+    FILES_MATCHING PATTERN "*.h*")
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/MaterialXRenderGlsl.pdb"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/" OPTIONAL)
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../../documents/Examples" DESTINATION ${CMAKE_INSTALL_PREFIX}/documents)
```
Comment (bernardkwok) [inline source/MaterialXRenderGlsl/CMakeLists.txt]: These 2 are installed multiple times. Should only install once. Actually I thought they would have been installed as part of shadergen. If not -- I think it should be added there.

File: source/MaterialXRenderHw/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+)
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/MaterialXRenderHw/
+    FILES_MATCHING PATTERN "*.h*")
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/MaterialXRenderHw.pdb"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/" OPTIONAL)
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../../documents/Examples" DESTINATION ${CMAKE_INSTALL_PREFIX}/documents)
```
Comment (bernardkwok) [inline source/MaterialXRenderHw/CMakeLists.txt]: Installed again.

File: source/MaterialXRenderOsl/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/MaterialXRenderOsl/
+    FILES_MATCHING PATTERN "*.h*")
+
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/MaterialXRenderOsl.pdb"
+        DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/" OPTIONAL)
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../../documents/Examples" DESTINATION ${CMAKE_INSTALL_PREFIX}/documents)
```
Comment (bernardkwok) [inline source/MaterialXRenderOsl/CMakeLists.txt]: Installed again.

File: source/MaterialXTest/CMakeLists.txt [hunk 5]
```diff
@@ -42,10 +42,14 @@ set_target_properties(
...
     SOVERSION "${MATERIALX_MAJOR_VERSION}")
 
-set (LIBS 
-    MaterialXCore 
-    MaterialXFormat 
+set (LIBS
+    MaterialXCore
+    MaterialXFormat
     MaterialXGenShader
+    MaterialXRender
```
Comment (bernardkwok) [inline source/MaterialXTest/CMakeLists.txt]: This is incorrect. This should be part of the conditional build below.

File: source/MaterialXTest/ShaderValid.cpp [hunk 6]
```diff
@@ -1112,7 +1112,7 @@ bool getTestOptions(const std::string& optionFile, ShaderValidTestOptions& optio
 
 void printRunLog(const ShaderValidProfileTimes &profileTimes, const ShaderValidTestOptions& options,
     std::set<std::string>& usedImpls, std::ostream& profilingLog, mx::DocumentPtr dependLib,
-    mx::OslShaderGeneratorPtr oslShaderGenerator, mx::GlslShaderGeneratorPtr glslShaderGenerator,
+    mx::ArnoldShaderGeneratorPtr oslShaderGenerator, mx::GlslShaderGeneratorPtr glslShaderGenerator,
```
Comment (bernardkwok) [inline source/MaterialXTest/ShaderValid.cpp]: This is incorrect: You should not revert this back to Arnold. There should be no arnold dependency in this test.

File: source/MaterialXTest/CMakeLists.txt (hunk 5 above)
Comment (bernardkwok) [reply source/MaterialXTest/CMakeLists.txt]: See line 63.

File: source/MaterialXTest/ShaderValid.cpp [hunk 7]
```diff
@@ -19,17 +19,17 @@
...
 #endif
 
 #ifdef MATERIALX_BUILD_GEN_OGSFX
 #include <MaterialXGenOgsFx/OgsFxShaderGenerator.h>
 #endif
 
 #ifdef MATERIALX_BUILD_GEN_OSL
-#include <MaterialXGenOsl/OslShaderGenerator.h>
-#include <MaterialXRender/ShaderValidators/Osl/OslValidator.h>
+#include <MaterialXGenArnold/ArnoldShaderGenerator.h>
```
Comment (bernardkwok) [inline source/MaterialXTest/ShaderValid.cpp]: This should be osl, not arnold.

File: source/MaterialXTest/ShaderValid.cpp [hunk 8]
```diff
@@ -1560,7 +1560,6 @@ TEST_CASE("MaterialX documents", "[shadervalid]")
                 if (nodeDef)
                 {
                     mx::string elementName = mx::replaceSubstrings(element->getNamePath(), pathMap);
-                    elementName = mx::createValidName(elementName);
```
Comment (bernardkwok) [inline source/MaterialXTest/ShaderValid.cpp]: This was lost.

File: source/MaterialXTest/ShaderValid.cpp (hunk 6 above)
Comment (feldstj) [reply source/MaterialXTest/ShaderValid.cpp]: Ya, it looks like somehow I merged things incorrectly, this whole file seems wrong.

**Pull request merged**